Networking/Networking/Model/Note.swift

```diff
@@ -238,3 +237,47 @@ extension Note {
         case storeReview = "store_review"
     }
 }
+
+extension Note {
+    public init?(payload: [String: Any]) throws {
+        guard let noteID = payload[CodingKeys.noteID.rawValue] as? Int64 else {
+            return nil
+        }
+        let hash: Int64 = payload[CodingKeys.hash.rawValue] as? Int64 ?? Int64.min
+        let read: Bool = (payload[CodingKeys.read.rawValue] as? Bool) ?? false
+        let icon: String? = payload[CodingKeys.icon.rawValue] as? String
+        let noticon: String? = payload[CodingKeys.noticon.rawValue] as? String
+        let timestamp: String = (payload[CodingKeys.timestamp.rawValue] as? String) ?? String()
+        let type: String = (payload[CodingKeys.type.rawValue] as? String) ?? String()
+        let subtype: String? = payload[CodingKeys.subtype.rawValue] as? String
+        let url: String? = payload[CodingKeys.url.rawValue] as? String
+        let title: String? = payload[CodingKeys.title.rawValue] as? String
+
+        let rawSubject = (payload[CodingKeys.subject.rawValue] as? [[String: Any]]) ?? []
+        let rawSubjectData = try? JSONSerialization.data(withJSONObject: rawSubject)
+
+        let rawHeader = (payload[CodingKeys.header.rawValue] as? [[String: Any]]) ?? []
+        let rawHeaderData = try? JSONSerialization.data(withJSONObject: rawHeader)
+
+        let rawBody = (payload[CodingKeys.body.rawValue] as? [[String: Any]]) ?? []
+        let rawBodyData = try? JSONSerialization.data(withJSONObject: rawBody)
+
+        let rawMeta = (payload[CodingKeys.meta.rawValue] as? [String: Any]) ?? [:]
+        let rawMetaData = try? JSONSerialization.data(withJSONObject: rawMeta)
+
+        self.init(noteID: noteID,
+                  hash: hash,
+                  read: read,
+                  icon: icon,
+                  noticon: noticon,
+                  timestamp: timestamp,
+                  type: type,
+                  subtype: subtype,
+                  url: url,
+                  title: title,
+                  subject: rawSubjectData ?? Data(),
+                  header: rawHeaderData ?? Data(),
+                  body: rawBodyData ?? Data(),
+                  meta: rawMetaData ?? Data())
+    }
+}
```

Comment:
To not duplicate and maintain this code twice, do you think we could do something like
```swift
let data = try JSONSerialization.data(withJSONObject: payload)
return try JSONDecoder().decode(Note.self, from: data)
```

That way we will only handle serialization in the custom decoding method

Reply:
I remembered now that I tried that and the problem are lines like:
`let hash = try container.decode(Int64.self, forKey: .hash)`
Only if I change `public init(from decoder: Decoder) throws {` to decode `hash` if there is some, and use `Int64.min` if there is no `hash` to be decoded? Since notification payload does not have `hash` for the note.

Reply:
That works, we also have `decodeIfPresent` method!